### DISCUSSION TITLE
Internationalization

### DISCUSSION THREAD

**Discussion by jpagex** [dillonkearns/elm-pages#292] [Ideas]
Hello,

Firstly, thanks a lot for this great project and all your efforts!

I was wondering if you already thought about internationalization. I don't think there is currently a way to configure a page to be accessible in multiple locales with a different URL. For example, if I have the page `src/Page/OurTeam.elm`, it would be nice if we could serve it under:
- `/en/our-team` for English
- `/fr/notre-equipe` for French
- `/es/nuestro-equipo` for Spanish
- …

We could even choose a default language where we can omit the locale prefix (e.g. `/our-team` if English is the default language).

At the moment, we can use a dynamic segment and handle that ourselves. But this means that we cannot benefit from the file-based routing possibilities.

Of course, this brings a lot of complexities and concerns (link to internal pages, SEO, translations, …). But it would be very beneficial for multilingual websites.

Let me know what you think about this topic.

Until next time… 🎶😉

**Comment by dillonkearns** [comment]
Hello @jpagex, thanks for the discussion! It's an interesting topic, I know that some other frameworks have arrived at the conclusion that the framework is the best place to provide some support for internationalized routing patterns. I think NextJS has a pretty thoughtful design there if I recall correctly.

I think a good place to start would be to look at some examples of how other frameworks approach the problem, and then try to imagine what similar designs in Elm would look like. That could get some ideas flowing.

Cheers!
